lib/Model/Acl.php

```diff
@@ -489,4 +443,170 @@ private function getIsShareValidForUsers(): bool {
 	private function getHasEmail(): bool {
 		return $this->share->getToken() ? strlen($this->share->getEmailAddress()) > 0 : $this->getIsLoggedIn();
 	}
+
+	/**
+	 * Checks, if user is allowed to edit the poll configuration
+	 **/
+	private function getAllowEditPoll(): bool {
+		// Console god mode
+		if (defined('OC_CONSOLE')) {
+			return true;
+		}
+
+		// owner is always allowed to edit the poll configuration
+		if ($this->getIsOwner()) {
+			return true;
+		}
+
+		// user has delegated owner rights
+		if ($this->getIsDelegatedAdmin()) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks, if user is allowed to access poll
+	 **/
+	private function getAllowAccessPoll(): bool {
+		// edit rights include access to poll
+		if ($this->getAllowEditPoll()) {
+			return true;
+		}
+
+		// No further access to poll, if it is deleted
+		if ($this->poll->getDeleted()) {
+			return false;
+		}
+
+		// grant access if user is involved in poll in any way
+		if ($this->getIsInvolved()) {
+			return true;
+		}
+
+		// grant access if poll poll is an open poll (for logged in users)
+		if ($this->poll->getAccess() === Poll::ACCESS_OPEN && $this->getIsLoggedIn()) {
+			return true;
+		}
+
+		// user has valid token of this poll
+		if ($this->getToken()) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks, if user is allowed to delete the poll
+	 * includes the right to archive and take over
+	 **/
+	private function getAllowDeletePoll(): bool {
+		// users with edit rights are allowed to delete the poll
+		if ($this->getAllowEditPoll()) {
+			return true;
+		}
+
+		// admins are allowed to delete the poll
+		if ($this->getIsAdmin()) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * User rights inside poll
+	 **/
+
+	/**
+	 * Checks, if user is allowed to add add vote options
+	 **/
+	private function getAllowAddOptions(): bool {
+		// Edit right includes adding new options
+		if ($this->getAllowEditPoll()) {
+			return true;
+		}
+
+		// deny, if user has no access right to this poll
+		if (!$this->getAllowAccessPoll()) {
+			return false;
+		}
+
+		// public shares are not allowed to add options
+		if ($this->share->getType() === Share::TYPE_PUBLIC) {
+			return false;
+		}
+
+		// Request for option proposals is expired, deny
+		if ($this->poll->getProposalsExpired()) {
+			return false;
+		}
+
+		return $this->poll->getAllowProposals() === Poll::PROPOSAL_ALLOW;
+	}
+
+	/**
+	 * Checks, if user is allowed to comment
+	 **/
+	private function getAllowComment(): bool {
+		// user has no access right to this poll
+		if (!$this->getAllowAccessPoll()) {
+			return false;
+		}
+
+		// public shares are not allowed to comment
+		if ($this->share->getType() === Share::TYPE_PUBLIC) {
+			return false;
+		}
+
+		return !!$this->poll->getAllowComment();
```

Comment:
That is very "JS" like. In PHP you would normally do:
```suggestion
		return (bool) $this->poll->getAllowComment();
```

but both works.

Reply:
Ah. Thanks!